Sources/SmithyReadWrite/SmithyWriter.swift

```diff
@@ -0,0 +1,47 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Foundation.Data
+import struct Foundation.Date
+import enum SmithyTimestamps.TimestampFormat
+
```

Comment:
Like the reader protocol above, `SmithyReader` is conformed to by the reader for all the supported readers (XML, JSON, & FormURL).

Concrete conforming types must implement methods to write all the types supported by Smithy models.